lib/pure/parsecfg.nim

```diff
@@ -359,3 +415,132 @@ proc next*(c: var CfgParser): CfgEvent {.rtl, extern: "npc$1".} =
+  var curSection = "" ## Current section, the default value of the current section is "", 
+                      ## which means that the current section is a common
+  var p: CfgParser
+  var fileStream = newFileStream(filename, fmRead)
+  if fileStream != nil:
+    open(p, fileStream, filename)
+    while true:
+      var e = next(p)
+      case e.kind
+      of cfgEof:
+        break
+      of cfgSectionStart: # Only look for the first time the Section
+        curSection = e.section
+      of cfgKeyValuePair:
+        var t = initOrderedTable[string, string]()
+        if dict.hasKey(curSection):
+          t = dict[curSection]
+        t[e.key] = e.value
+        dict[curSection] = t
+      of cfgOption:
+        var c = initOrderedTable[string, string]()
+        if dict.hasKey(curSection):
+          c = dict[curSection]
+        c["--" & e.key] = e.value
+        dict[curSection] = c
+      of cfgError:
+        break
+    close(p)
+  result = dict
+
+proc hasNotSymchars(s: string): bool =
+  result = false
+  for ch in s:
+    if not (ch in SymChars):
+      result = true
+      break
+
+proc replace(s: string): string =
+  var d = ""
+  var i = 0
+  while i < s.len():
+    if s[i] == '\\':
+      d.add(r"\\")
+    elif s[i] == '\c' and s[i+1] == '\L':
+      d.add(r"\n")
+      inc(i)
+    elif s[i] == '\c':
+      d.add(r"\n")
+    elif s[i] == '\L':
+      d.add(r"\n")
+    else:
+      d.add(s[i])
+    inc(i)
+  result = d
+
+proc writeCfgFile*(dict: OrderedTable[string, OrderedTable[string, string]], filename: string) =
+  ## Writes the contents of the table to the specified configuration file.
+  ## Note: write to ignore the comment
+  var file = open(filename, fmWrite)
+  var section, key, value, kv, segmentChar:string
+  for pair in dict.pairs():
+    section = pair[0]
+    if section != "": ## Not general section
+      if hasNotSymchars(section): ## Non system character
+        file.writeLine("[\"" & section & "\"]")
+      else:
+        file.writeLine("[" & section & "]")
+    for pair2 in pair[1].pairs():
+      key = pair2[0]
+      value = pair2[1]
+      if key[0] == '-' and key[1] == '-': ## If it is a command key
+        segmentChar = ":"
+        if hasNotSymchars(key[2..key.len()-1]):
+          kv = "--\"" & key[2..key.len()-1] & "\""
+        else:
+          kv = key
+      else:
+        segmentChar = "="
+        kv = key
+      if value != "": ## If the key is not empty
+        if hasNotSymchars(value):
+          kv = kv & segmentChar & "\"" & replace(value) & "\""
+        else:
+          kv = kv & segmentChar & value
+      file.writeLine(kv)
+  file.close()
+
+proc getSectionKey*(dict: OrderedTable[string, OrderedTable[string, string]], section, key: string): string =
+  ## Gets the Key value of the specified Section for a given table.
+  ## Before using it to call the loadCfgFile function.
+  result = ""
+  if dict.haskey(section):
+    if dict[section].hasKey(key):
+      result = dict[section][key]
```

Comment:
Move the 'result = ""' down into the else (or use a 'return') to save the string allocation in case it has the key.